tests/suite-unit/PluginFlyvemdmPolicyRemoveapplication.php

```diff
@@ -0,0 +1,233 @@
+   public function unicityCheckProvider() {
+      return [
+         'Check for invalid package'            => [
+            'data'     => ['package_name', 'id'],
+            'expected' => [true],
+         ],
+         // TODO: For test this case a fleet must be created first
+         /*'Check for valid package' => [
+            'data'     => ['package_name', 'id'],
+            'expected' => [false],
+         ],*/
+      ];
+   }
+
+   /**
+    * @dataProvider unicityCheckProvider
+    * @tags testUnicityCheck
+    * @param array $data
+    * @param array $expected
+    */
+   public function testUnicityCheck(array $data, array $expected) {
+      list($policy) = $this->createNewPolicyInstance();
+      $mockInstance = $this->newMockInstance('\PluginFlyvemdmFleet');
+      $mockInstance->getMockController()->getID = 1;
+      if(!$expected[0]){
+         $application = $this->createAppInDB();
+         $data[0] = $application->getName();
+         $data[1] = $application->getID();
+      }
+      $this->boolean($policy->unicityCheck($data[0], \PluginFlyvemdmPackage::class, $data[1],
+         $mockInstance))->isEqualTo($expected[0]);
+   }
+
+   /**
+    * @return array
+    */
+   public function applyProvider() {
+      $package = $this->createAppInDB();
+      return [
+         'Check for invalid arguments 1' => [
+            'data'     => [null, null, null],
+            'expected' => ['return' => false],
+         ],
+         'Check for invalid arguments 2' => [
+            'data'     => [null, null, ''],
+            'expected' => ['return' => false, 'message'=>'An application ID is required'],
+         ],
+         'Check for invalid arguments 3' => [
+            'data'     => [null, null, -10],
+            'expected' => ['return' => false, 'message'=>'The application does not exists'],
+         ],
+         'Check for correct value' => [
+            'data'     => [null, $package, $package->getID()],
+            'expected' => ['return' => true],
+         ],
+      ];
+   }
+
+   /**
+    * @dataProvider applyProvider
+    * @tags testApply
+    * @param array $data
+    * @param array $expected
+    */
+   public function testApply(array $data, array $expected) {
+      $policyData = null;
+      if (true === $expected['return']) {
+         $policyData = new \PluginFlyvemdmPolicy();
+         $policyData->getFromDBBySymbol('removeApp');
+      }
+      list($policy) = $this->createNewPolicyInstance($policyData);
+      $mockInstance = $this->newMockInstance('\PluginFlyvemdmFleet');
+      $mockInstance->getMockController()->getID = 1;
+      $result = $policy->apply($mockInstance, $data[0], $data[1], $data[2]);
+      $this->boolean($result)->isEqualTo($expected['return']);
+      if (!$expected['return'] && isset($expected['message'])) {
+         $this->string($_SESSION["MESSAGE_AFTER_REDIRECT"][1][0])->isEqualTo($expected['message']);
+         unset($_SESSION["MESSAGE_AFTER_REDIRECT"]); // to clear the buffer
```

Comment:
Coming again here.

the unset() is done only when the test method is successful. If it fails, it may break an other execution of the method for the next data set from the data provider.

in the if(), copy ```$_SESSION["MESSAGE_AFTER_REDIRECT"][1][0]``` in a variable, next unset it, next test the variable.